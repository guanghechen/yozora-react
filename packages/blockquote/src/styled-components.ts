import 'styled-components'
import type { YozoraBlockquoteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    blockquote?: YozoraBlockquoteTheme
  }
}
