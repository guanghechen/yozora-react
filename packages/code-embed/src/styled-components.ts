import type { YozoraCodeEmbedTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeEmbed?: YozoraCodeEmbedTheme
  }
}
