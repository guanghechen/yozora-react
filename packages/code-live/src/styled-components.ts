import 'styled-components'
import type { YozoraCodeLiveTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeLive?: YozoraCodeLiveTheme
  }
}
