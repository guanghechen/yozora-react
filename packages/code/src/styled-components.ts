import 'styled-components'
import type { YozoraCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    code?: YozoraCodeTheme
  }
}
