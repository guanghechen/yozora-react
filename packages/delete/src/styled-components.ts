import type { YozoraDeleteTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    delete?: YozoraDeleteTheme
  }
}
