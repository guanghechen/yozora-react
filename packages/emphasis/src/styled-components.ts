import type { YozoraEmphasisTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    emphasis?: YozoraEmphasisTheme
  }
}
