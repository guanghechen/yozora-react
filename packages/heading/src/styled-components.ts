import 'styled-components'
import type { YozoraHeadingTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    heading?: YozoraHeadingTheme
  }
}
