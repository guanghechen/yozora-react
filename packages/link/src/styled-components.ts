import type { YozoraLinkTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    link?: YozoraLinkTheme
  }
}
