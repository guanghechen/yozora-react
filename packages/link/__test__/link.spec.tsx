import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import Link from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('render a simple content', () => {
    const text = 'Hello, world!'
    const wrapper = render(
      <Router>
        <Link url="#">{ text }</Link>
      </Router>
    )
    expect(wrapper.text()).toEqual(text)
  })

  it('render with custom className', () => {
    const text = 'Hello, world!'
    const className = 'custom-link'
    const wrapper = render(
      <Router>
        <Link url="https://www.github.com" className={ className }>
          <span>{ text }</span>
        </Link>
      </Router>
    )
    expect(wrapper.hasClass(className)).toEqual(true)
    expect(wrapper.text()).toEqual(text)
  })

  it('url and children both are required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(
          <Router>
            <Link url={ value }>link text</Link>
          </Router>
        )
      }).toThrow(/Failed prop type: The prop `url` is marked as required/i)

      expect(() => {
        render(
          <Router>
            <Link url="/home">{ value }</Link>
          </Router>
        )
      }).toThrow(/Failed prop type: The prop `children` is marked as required/i)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLAnchorElement>()
    const wrapper = mount(
      <Router>
        <Link
          ref={ ref }
          url="https://www.github.com"
          data-value="waw"
        >
          1
        </Link>
      </Router>
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const wrapper = render(
      <Router>
        <Link
          url="/home"
          title="home"
          style={{ color: 'orange', fontSize: '16px' }}
        >
          some text1
          <span>some text2</span>
        </Link>
      </Router>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      yozora: {
        link: {
          color: 'blue',
          fontSize: undefined,
          fontStyle: 'italic',
          textDecoration: 'none',
        }
      }
    }

    const wrapper = mount(
      <Router>
        <ThemeProvider theme={ theme }>
          <Link
            url="https://www.github.com"
            title="home"
            style={{ color: 'orange', fontSize: '16px' }}
          >
            some text1
            <span>some text2</span>
          </Link>
        </ThemeProvider>
      </Router>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
