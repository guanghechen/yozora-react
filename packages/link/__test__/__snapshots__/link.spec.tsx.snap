// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
.c0 {
  color: inherit;
  font-size: inherit;
  font-style: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  class="c0"
  href="/home"
  style="color:orange;font-size:16px"
  theme="[object Object]"
  title="home"
>
  some text1
  <span>
    some text2
  </span>
</a>
`;

exports[`basic rendering case snapshot with theme 1`] = `
.c0 {
  color: blue;
  font-size: inherit;
  font-style: italic;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component
      theme={
        Object {
          "yozora": Object {
            "link": Object {
              "color": "blue",
              "fontSize": undefined,
              "fontStyle": "italic",
              "textDecoration": "none",
            },
          },
        }
      }
    >
      <YozoraLink
        style={
          Object {
            "color": "orange",
            "fontSize": "16px",
          }
        }
        title="home"
        url="https://www.github.com"
      >
        <YozoraExternalLink
          style={
            Object {
              "color": "orange",
              "fontSize": "16px",
            }
          }
          title="home"
          url="https://www.github.com"
        >
          <styled.a
            href="https://www.github.com"
            rel="noreferrer"
            style={
              Object {
                "color": "orange",
                "fontSize": "16px",
              }
            }
            target="_blank"
            theme={
              Object {
                "yozora": Object {
                  "link": Object {
                    "color": "inherit",
                    "fontSize": "inherit",
                    "fontStyle": "inherit",
                    "textDecoration": "none",
                  },
                },
              }
            }
            title="home"
          >
            <a
              className="c0"
              href="https://www.github.com"
              rel="noreferrer"
              style={
                Object {
                  "color": "orange",
                  "fontSize": "16px",
                }
              }
              target="_blank"
              title="home"
            >
              some text1
              <span>
                some text2
              </span>
            </a>
          </styled.a>
        </YozoraExternalLink>
      </YozoraLink>
    </Component>
  </Router>
</BrowserRouter>
`;
